src/python/pants/backend/scala/bsp/rules.py

```diff
@@ -114,7 +115,7 @@ class ThirdpartyModulesRequest:
 @dataclass(frozen=True)
 class ThirdpartyModules:
     resolve: CoursierResolveKey
-    entries: dict[CoursierLockfileEntry, ClasspathEntry]
+    entries: dict[CoursierLockfileEntry, tuple[ClasspathEntry, list[CoursierLockfileEntry]]]
```

Comment:
The inner `list` should be a `tuple` instead since "frozen" dataclasses need to be hashable and `list` is not hashable (since it is mutable).

Reply:
Also, maybe it would be better to just use a new dataclass instead of a 2-tuple? Then you could give a name to each component which will help with readability.

Reply:
And is there any particular ordering of these `CoursierLockfileEntry` instances?